src/App.vue

```diff
@@ -325,31 +321,39 @@ export default class App extends Mixins(BcolMixin, DateMixin, FilingTemplateMixi
       // reset errors in case this method is invoked more than once (ie, retry)
       this.resetFlags()
 
-      // get business ID query param
-      const businessId = this.$route?.query?.businessId as string
-      if (!businessId) {
-        this.fetchErrorDialog = true
-        if (!businessId) throw new Error('Invalid business identifier')
-      }
-      this.setBusinessId(businessId)
-
       // ensure user is authorized or is staff to access this business
-      await this.checkAuth(businessId).catch(error => {
+      await this.checkAuth(this.getBusinessId).catch(error => {
         this.accountAuthorizationDialog = true
         throw new Error(`Auth error: ${error}`)
       })
 
-      // fetch IA filing to alter or correct
-      const { filing } = await this.fetchFiling(this.INCORPORATION_APPLICATION) // TEMP FOR TESTING
-      // const { filing } = await this.fetchFiling(this.ALTERATION) // FUTURE STATE
-
-      // parse filing into store
-      this.parseIncorpFiling(filing)
+      // check for necessary query params
+      switch (this.$route.name) {
+        case RouteNames.ALTERATION:
+          // const filingId = +this.$route.query['filing-id']
+          // if (isNaN(filingId)) {
+          //   this.fetchErrorDialog = true
+          //   throw new Error('Invalid alteration filing ID')
+          // }
+          this.setEntityType(EntityTypes.BCOMP)
+          break
+        case RouteNames.CORRECTION:
+          const correctedId = +this.$route.query['corrected-id']
+          const correctionId = +this.$route.query['correction-id']
+          if (isNaN(correctedId) && isNaN(correctionId)) {
+            this.fetchErrorDialog = true
+            throw new Error('Invalid corrected or correction filing ID')
+          }
+          this.setEntityType(EntityTypes.BCOMP)
+          break
+      }
 
+      // TODO: move this to views?
```

Comment:
Let me know your thoughts. I'd prefer to keep App.vue as simple as possible.

Reply:
I definitely agree! I think it's a good idea to move this to the views, or at least call the method from the views. I believe that functionality would be a good mixin? Would need a little refactor. 
That being said, would you want the calculator itself in the views or just have the views pass the appropriate route name or whatever other identifier back to the app?
If the calculator is in the views separately we will have some duplication there.

Reply:
Umm, I dunno. We can think about it later!

Reply:
Absolutely!

Reply:
I moved this to the views after all, along with an event for the view to update App.